youtube_dl/extractor/douyutv.py

```diff
@@ -75,74 +73,74 @@ def _real_extract(self, url):
             room_id = self._html_search_regex(
                 r'"room_id"\s*:\s*(\d+),', page, 'room id')
 
-        config = None
+        room_url = 'http://m.douyu.com/html5/live?roomId=%s' % room_id
+        room_content = self._download_webpage(room_url, video_id)
+        room_json = self._parse_json(room_content, video_id, fatal=False)
+
+        room = room_json['data']
+
+        show_status = room.get('show_status')
+        # 1 = live, 2 = offline
+        if show_status == '2':
+            raise ExtractorError(
+                'Live stream is offline', expected=True)
+
+        flv_json = None
         # Douyu API sometimes returns error "Unable to load the requested class: eticket_redis_cache"
         # Retry with different parameters - same parameters cause same errors
         for i in range(5):
-            prefix = 'room/%s?aid=android&client_sys=android&time=%d' % (
-                room_id, int(time.time()))
-            auth = hashlib.md5((prefix + '1231').encode('ascii')).hexdigest()
+            tt = int(time.time() / 60)
+            did = uuid.uuid4().hex.upper()
+
+            sign_content = '{room_id}{did}A12Svb&%1UUmf@hC{tt}'.format(room_id = room_id, did = did, tt = tt)
```

Comment:
How did you get this key? Better to add some comments so that others can follow it if the key is changed.